rosdep keys used by noetic on Ubuntu 22.04

dreuter opened the pull request:
Please add the following dependency to the rosdep database.

## Package name:

- hddtemp
- liborocos-kdl
- libogre[-dev]

## Package Upstream Source:

omitted since they are already present in the `base.yml`

## Purpose of using this:

I am trying to build noetic on Ubuntu 22.04. This allows me to install the dependencies as stated in the ["Installing from source"](http://wiki.ros.org/noetic/Installation/Source) instructions.

Distro packaging links:



- Debian: https://packages.debian.org/
  - Not changed
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/jammy/liborocos-kdl1.5
  - https://packages.ubuntu.com/jammy/hddtemp (see also https://packages.ubuntu.com/focal/hddtemp)
  - https://packages.ubuntu.com/jammy/libogre-1.9.0v5
  - https://packages.ubuntu.com/jammy/libogre-1.9-dev
- Fedora: https://src.fedoraproject.org/browse/projects/
  - Not changed
- Arch: https://www.archlinux.org/packages/
  - Not changed
- Gentoo: https://packages.gentoo.org/
  - Not changed
- macOS: https://formulae.brew.sh/
  - Not changed
- Alpine: https://pkgs.alpinelinux.org/packages
  - Not changed
- NixOS/nixpkgs: https://search.nixos.org/packages
  - Not changed

dreuter committed:
add `libogre` and `libogre-dev` for `jammy`

These are the only two rosdep keys that are missing to run `rosdep` on
`noetic` `desktop-full` on Ubuntu 22.04, as described in the
["Installing from source"](http://wiki.ros.org/noetic/Installation/Source)
tutorial.

dreuter committed:
update `liborocos-kdl` for jammy

On Ubuntu 22.04 only version 1.5 of `liborocos-kdl` is available, but
noetic seems to compile fine with this version.

dreuter committed:
"disable" `hddtemp` on jammy

On Ubuntu 22.04 `hddtemp` seems to be no longer released (See
[packages.ubuntu.com](https://packages.ubuntu.com/focal/hddtemp), there
is no link to jammy).

Therefore to not run into any issues when using `rosdep` I have disabled
the installation of `hddtemp` for Ubuntu 22.04.

methylDragon commented on rosdep/base.yaml:
```diff
@@ -4328,6 +4332,7 @@ liborocos-kdl:
     '*': [liborocos-kdl1.4]
     bionic: [liborocos-kdl1.3]
     xenial: [liborocos-kdl1.3]
+    jammy: [liborocos-kdl1.5]
```
I think CI is failing because this isn't alphabetically sorted?

dreuter replied:
Done through change requested below :)

tfoote commented on rosdep/base.yaml:
```diff
@@ -4328,6 +4332,7 @@ liborocos-kdl:
     '*': [liborocos-kdl1.4]
     bionic: [liborocos-kdl1.3]
     xenial: [liborocos-kdl1.3]
+    jammy: [liborocos-kdl1.5]
```
Kinetic is also targeting 1.5. We should move the wildcard key to 1.5 and list out the old 1.4 versions explicitly instead of just stating Jammy so that we can keep using the wildcard hopefully in future releases.

dreuter replied:
done

dreuter committed:
"disable" `hddtemp` by default

Currently `hddtemp` is only available on `bionic`, `focal` and `impish`.
Since it is unclear whether it will be released in a future version of
Ubuntu the default will now be to disable `hddtemp` on all not
explicitly mentioned releases.

dreuter committed:
make `liborocos-kdl1.5` the default for `liborocos-kdl`

Since `liborocos-kdl1.5` will also most likely be the right package in
Ubuntu `kinetic` we make this the default and explicitly mention
`focal` and `impish` for
[`liborocos-kdl1.4`](https://packages.ubuntu.com/focal/liborocos-kdl1.4).

dreuter commented on rosdep/base.yaml:
```diff
@@ -1739,7 +1739,9 @@ hddtemp:
-  ubuntu: [hddtemp]
+  ubuntu:
+    "*": [hddtemp]
+    jammy: null
```
I am not sure if it is going away or if it is just not released yet/in jammy.
However I did mention `focal`, `jammy` and impish explicitly now and disabled it on all other releases.

tfoote commented on rosdep/base.yaml:
```diff
@@ -4133,6 +4138,7 @@ libogre-dev:
     artful: [libogre-1.9-dev]
     bionic: [libogre-1.9-dev]
     focal: [libogre-1.9-dev]
+    jammy: [libogre-1.9-dev]
```
You missed my suggestion about turning this into a wildcard. However, actually looking at this all versions back to xenial are `[libogre-1.9-dev]` so we can just collapse this. to be an unversioned rule as they're all the same now.

dreuter replied:
The same is true for `libogre` itself: https://packages.ubuntu.com/jammy/libogre-1.9.0v5
(libogre-dev: https://packages.ubuntu.com/jammy/libogre-1.9-dev)
Should I collapse this too?

dreuter replied:
I have changed both, please tell me if I should change it :)

tfoote reviewed (changes requested):
Thanks for the updates. The libogre-dev rule needs one small update still.

dreuter committed:
collapse ubuntu mapping for `libogre[-dev]`

All releases of Ubuntu since xenial use the same package name for
`libogre[-dev]`, therefore we can collapse the mapping.

dreuter committed:
collapse ubuntu mapping for `libfltk-dev`

All releases of Ubuntu since xenial use the same package name for
[`libfltk-dev`](https://packages.ubuntu.com/jammy/libfltk1.3-dev),
therefore we can collapse the mapping.
This has the added benefit that this key will also work on `jammy` and
`kinetic`.

nuclearsandwich commented:
> I can also create a separate PR for this if you want :)

In general more focused pull requests are always easier to review.

ivanpauno reviewed (approved):
LGTM!

tfoote reviewed (approved):
Thanks for the updates. It's great to get things more compact.

tfoote reviewed (commented):
@cottsay I'm not sure why this is failing outside of the changed section. Clearly this rule is incorrect, but it shouldn't be checking here for this PR.

```
Check failure on line 707 in rosdep/base.yaml

GitHub Actions
/ Nosetests (3.8)

rosdep/base.yaml#L707

Package 'libcurlpp-dev' could not be found for debian buster on amd64
```

cottsay commented:
> I'm not sure why this is failing outside of the changed section.

This misbehavior is described in #31439. In short, #33130 removed a line that wasn't passing the check, and when the test diffs the HEAD branch against `origin/master`, it sees the line as being added because the HEAD branch doesn't contain the change to remove it.

Rebasing or merging from master into the HEAD of this PR will make the check pass.

nuclearsandwich commented:
> Rebasing or merging from master into the HEAD of this PR will make the check pass.

@dreuter I took the liberty of pushing a rebase.

dreuter commented:
> > Rebasing or merging from master into the HEAD of this PR will make the check pass.
> 
> @dreuter I took the liberty of pushing a rebase.

Thank you very much :)
